// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message MonitoringMetricsScope {
  string name = 1;
  string create_time = 2;
  string update_time = 3;
  repeated MonitoringMetricsScopeMonitoredProjects monitored_projects = 4;
}

message MonitoringMetricsScopeMonitoredProjects {
  string name = 1;
  string create_time = 2;
}

message ApplyMonitoringMetricsScopeRequest {
  MonitoringMetricsScope resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteMonitoringMetricsScopeRequest {
  string service_account_file = 1;
  MonitoringMetricsScope resource = 2;
}

message ListMonitoringMetricsScopeRequest {
  string service_account_file = 1;
}

message ListMonitoringMetricsScopeResponse {
  repeated MonitoringMetricsScope items = 1;
}

service MonitoringMetricsScopeService {
  rpc ApplyMonitoringMetricsScope(ApplyMonitoringMetricsScopeRequest) returns (MonitoringMetricsScope);
  rpc DeleteMonitoringMetricsScope(DeleteMonitoringMetricsScopeRequest) returns (google.protobuf.Empty);
  rpc ListMonitoringMetricsScope(ListMonitoringMetricsScopeRequest) returns (ListMonitoringMetricsScopeResponse);
}
